src/Contrib/Jaeger/JaegerTransport.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Jaeger;
+
+use Jaeger\Thrift\Agent\AgentClient;
+use Jaeger\Thrift\Batch;
+use Jaeger\Thrift\Process;
+use Jaeger\Thrift\Span;
+use Thrift\Exception\TTransportException;
+use Thrift\Protocol\TCompactProtocol;
+
+final class JaegerTransport implements Transport
+{
+
+    // DEFAULT_BUFFER_SIZE indicates the default maximum buffer size, or the size threshold
+    // at which the buffer will be flushed to the agent.
+    const DEFAULT_BUFFER_SIZE = 1;
+
+    private $transport;
+    private $client;
+
+    private $buffer = [];
+    private $process = null;
+    private $maxBufferSize = 0;
+
+    public function __construct($address = '127.0.0.1', $port = 6831, $maxBufferSize = 0)
+    {
+        $this->transport = new ThriftUdpTransport($address, $port);
+        $p = new TCompactProtocol($this->transport);
+        $this->client = new AgentClient($p, $p);
+
+        $this->maxBufferSize = ($maxBufferSize > 0 ? $maxBufferSize : self::DEFAULT_BUFFER_SIZE);
+    }
+
+    /**
+    * Submits a new span to collectors, possibly delayed and/or with buffering.
+    *
+    * @param Span $span
+    */
+    public function append(Span $span, $serviceName)
+    {
+        // Grab a copy of the process data, if we didn't already.
+        if ($this->process == null) {
+            $this->process = new Process([
+                'serviceName' => $serviceName,
+                'tags' => $span->tags,
+            ]);
+        }
+
+        $this->buffer[] = $span;
+
+        // TODO(tylerc): Buffer spans and send them in as few UDP packets as possible.
+        return $this->flush();
+    }
+
+    /**
+    * Flush submits the internal buffer to the remote server. It returns the
+    * number of spans flushed.
+    *
+    * @param bool $force - force a flush, even on a partial buffer
+    */
+    public function flush($force = false)
+    {
+        $spans = count($this->buffer);
```

Comment:
type-hint the return type (int)... (if `flush()` does indeed need to return this data, it's not used, so maybe it should return `void` instead)

Reply:
Is there a way for one of the static analysis or code styling tools to let me know all the places that are missing return types? I'll try to go through and get them all just want to make sure I don't miss something

Reply:
Updated as many as I noticed were missing (left the ones extending the abstract class from the thrift generated files as is)